darts/models/forecasting/transformer_model.py

```diff
@@ -276,47 +278,118 @@ def __init__(
+
+        # start token consists only of target series, past covariates are substituted with 0 padding
+        start_token = self._permute_transformer_inputs(data[:, -1:, : self.target_size])
+        start_token_padded = F.pad(start_token, pad_size)
+
+        if len(x_in) == 3:
+            src, _, tgt = x_in
+            src = self._permute_transformer_inputs(src)
+            tgt_permuted = self._permute_transformer_inputs(tgt)
+            tgt_padded = F.pad(tgt_permuted, pad_size)
+            tgt = torch.cat([start_token_padded, tgt_padded], dim=0)
+            return self._prediction_step(src, tgt)[:, :-1, :, :]
+
         data, _ = x_in
-        # Here we create 'src' and 'tgt', the inputs for the encoder and decoder
-        # side of the Transformer architecture
-        src, tgt = self._create_transformer_inputs(data)
 
+        src = self._permute_transformer_inputs(data)
+        tgt = start_token_padded
+
+        predictions = []
+        for _ in range(self.target_length):
+            pred = self._prediction_step(src, tgt)[:, -1, :, :]
+            predictions.append(pred)
+            tgt = torch.cat(
+                [tgt, F.pad(pred.mean(dim=2).unsqueeze(dim=0), pad_size)],
+                dim=0,
+            )  # take average of quantiles
+        return torch.stack(predictions, dim=1)
+
+    def _prediction_step(self, src: torch.Tensor, tgt: torch.Tensor):
+        target_length = tgt.shape[0]
+        device, tensor_type = src.device, src.dtype
         # "math.sqrt(self.input_size)" is a normalization factor
         # see section 3.2.1 in 'Attention is All you Need' by Vaswani et al. (2017)
-        src = self.encoder(src) * math.sqrt(self.input_size)
-        src = self.positional_encoding(src)
+        src = self.encoder(src) * math.sqrt(self.d_model)
+        tgt = self.encoder(tgt) * math.sqrt(self.d_model)
 
-        tgt = self.encoder(tgt) * math.sqrt(self.input_size)
+        src = self.positional_encoding(src)
         tgt = self.positional_encoding(tgt)
 
-        x = self.transformer(src=src, tgt=tgt)
+        tgt_mask = torch.triu(
+            torch.full(
+                (target_length, target_length),
+                float("-inf"),
+                device=device,
+                dtype=tensor_type,
+            ),
+            diagonal=1,
+        )
+
+        x = self.transformer(src=src, tgt=tgt, tgt_mask=tgt_mask)
         out = self.decoder(x)
 
         # Here we change the data format
         # from (1, batch_size, output_chunk_length * output_size)
         # to (batch_size, output_chunk_length, output_size, nr_params)
-        predictions = out[0, :, :]
+        predictions = self._permute_transformer_inputs(out)
         predictions = predictions.view(
-            -1, self.target_length, self.target_size, self.nr_params
+            -1, target_length, self.target_size, self.nr_params
         )
 
         return predictions
 
+    # Allow teacher forcing
```

Comment:
I would remove this comment, and add in the method docstring that this rely on teacher forcing (eventually with a link to some resources/documentation/section in the original article)